people-and-planet-ai/land-cover-classification/create_dataset.py

```diff
@@ -0,0 +1,233 @@
+"""
+
+from typing import Iterable, List, Optional, Tuple
+
+import apache_beam as beam
+from apache_beam.options.pipeline_options import PipelineOptions
+import ee
+import numpy as np
+
+from serving import data
+
+# Default values.
+POINTS_PER_CLASS = 100
+PATCH_SIZE = 128
+MAX_REQUESTS = 20  # default EE request quota
+
+# Type aliases.
+Point = Tuple[float, float]  # (lon, lat)
+Polygon = List[Point]
+Example = Tuple[np.ndarray, np.ndarray]
+
+# Simplified polygons covering most land areas in the world.
+WORLD_POLYGONS = [
+    # Americas
+    [(-33.0, -7.0), (-55.0, 53.0), (-166.0, 65.0), (-68.0, -56.0)],
+    # Africa, Asia, Europe
+    [
+        (74.0, 71.0),
+        (166.0, 55.0),
+        (115.0, -11.0),
+        (74.0, -4.0),
+        (20.0, -38.0),
+        (-29.0, 25.0),
+    ],
+    # Australia
+    [(170.0, -47.0), (179.0, -37.0), (167.0, -12.0), (128.0, 17.0), (106.0, -29.0)],
+]
+
+
+def sample_points(
+    seed: int, polygons: List[Polygon], points_per_class: int, scale: int = 1000
+) -> Iterable[Point]:
+    """Selects around the same number of points for every classification.
+
+    This expects the input image to be an integer, for balanced regression points
+    you could do `image.int()` to truncate the values into an integer.
+    If the values are too large, it might be good to bucketize, for example
+    the range is between 0 and ~1000 `image.divide(100).int()` would give ~10 buckets.
+
+    Args:
+        seed: Random seed to make sure to get different results on different workers.
+        polygons: List of polygons to pick points from.
+        points_per_class: Number of points per classification to pick.
+        scale: Number of meters per pixel, a smaller scale will take longer.
+
+    Returns: An iterable of the picked points.
+    """
+    data.ee_init()
+    image = data.get_label_image()
+    region = ee.Geometry.MultiPolygon(polygons)
+    points = image.stratifiedSample(
+        points_per_class,
+        region=region,
+        scale=scale,
+        seed=seed,
+        geometries=True,
+    )
+    for point in points.toList(points.size()).getInfo():
+        yield point["geometry"]["coordinates"]
+
+
+def get_training_example(point: Point, patch_size: int = PATCH_SIZE) -> Example:
+    """Gets an (inputs, labels) training example for year 2020.
+
+    Args:
+        point: A (longitude, latitude) pair.
+        patch_size: Size in pixels of the surrounding square patch.
+
+    Returns: An (inputs, labels) pair of NumPy arrays.
+    """
+    data.ee_init()
+    return (
+        data.get_input_patch(2020, point, patch_size),
+        data.get_label_patch(point, patch_size),
+    )
+
+
+def try_get_example(point: Point, patch_size: int = PATCH_SIZE) -> Iterable[Example]:
+    """Wrapper over `get_training_examples` that allows it to simply log errors instead of crashing."""
+    try:
+        yield get_training_example(point, patch_size)
+    except Exception as e:
```

Comment:
Ditto

Reply:
Sure, I think `requests.exceptions.RequestException` should be the only thing unless an incorrect inputs are passed.